### DISCUSSION TITLE
Showing progress during long action call

### DISCUSSION THREAD

**Discussion by sebreunion** [livewire/livewire#4477] [1. Help] [answered]
Hi all,

I have some long code that runs inside a component action call. I would like to refresh the UI with some sort of progress indicator.
I set a public property in which i store the progress and i'd like to display this property on the UI every time this property change even if the action call is not over.

```
for ($i=0; $i < 3; $i++) {
    $this->progress = "XXXX";
    $this->dispatchBrowserEvent('progress-changed',['progress'=>$this->progress]);
    sleep(1);
}
```
```
 <h3 @progress-changed.window="alert('XXX'); progress = $event.detail.progress" x-text="progress"></h3>
```

I tried to use alpine.js and manage to update the UI but only when my action call code is complete, so the Ui is refreshed only once, not each time i set the property => I'm getting the 3 alerts back to back.

I tried the wire:poll but i would like to prevent spamming the server as i haven't found a way to trigger the wire:poll on a certain action call (it fires when the page load) and i'm not sure that it would work updating my UI during my long action call.

What would be the best way to update tue UI **during** a long action call ?

Thx

**Comment by sebreunion** [comment]
More digging....

Tried wire:poll but it wait until my long action is finished so it does not refresh the UI during the time the long call is running.

Even tried to change my function so that i no longer run my code on an entire collection but item by item. 

Manage in my blade to call 

`@this.call('myFunction', id);
`

like 50 times in a row, one time per item. 

My function is called 50 times but the component is not refresh after each despite the render() function being called. It is confirmes as In this case, i only see one ajax call in the chrome debug bar.

The only way i found is to work with an array of promises and synchronous call, this way the UI is refreshed after each call so i can display a progress message but, as expected with 50 ajax call instead of 1, the performance is quite poor...

**Comment by jonneroelofs** [comment, marked as answer]
You could consider moving the code that takes so much time to a job. You could then wire:poll and check how many jobs are left in the queue.

**Reply by sebreunion** [reply]
> You could consider moving the code that takes so much time to a job. You could then wire:poll and check how many jobs are left in the queue.

Yes, i think it's the way to go.

I'll give it a try.

Thx
